### PR TITLE
Closes #1219: fixes a 'null' toast on autofill failure on Android 11

### LINKED ISSUE #1219: Disable "null" toast
## Steps to reproduce

- Set Lockwise as Autofill Provider on Android 11
- Tap on a textbox that doesn't need Autofill (I'm not sure on this part, happens on a lot of textboxes).

### Expected behavior

Nothing happens

### Actual behavior

A toast appears with text "null".

### Device & build information
* Device: Pixel 2 XL (taimen)
* Build version: 4.0.1
### Notes

Another person could reproduce it at https://www.reddit.com/r/android_beta/comments/hs0d6k/null_toast/fyqnoox

Attachments:

### PR DESCRIPTION
**Pull request by severinrudie**
Fixes #1219

## Testing and Review Notes

- Use device running Android 11 (I was unable to repro on emulator)
- Install master Lockwise
- Set Lockwise as autofill provider
- Close and open Play Store app
- Select search bar at top of screen
--> EXPECT toast with text "null"
- Install `1219-fix-null-toast` Lockwise
- Close and open Play Store app
- Select search bar at top of screen
--> EXPECT no toast

## To Do

- [X] add “WIP” to the PR title if pushing up but not complete nor ready for review
- [X] double check the original issue to confirm it is fully satisfied
- [X] add testing notes and screenshots in PR description to help guide reviewers
- [ ] add unit tests
  - Unfeasible
- consider running this branch in a debug simulator and check for memory leak notifications or any warnings
- [ ] request the "UX" team perform a design review (if/when applicable)
  - Not applicable, no UI changes
- [ ] make sure CI builds are passing (e.g.: fix lint and other errors)
- [X] check on the [accessibility](https://mozilla-lockwise.github.io/lockwise-android/accessibility/) of any added UI
  - None

### CODE REVIEW

**Comment by Baron-Severin** [comment]
I do have one open question here. I'm not clear on what calling callback.onFailure is supposed to do.  I didn't notice any difference on testing.  If it's important, we could do a version check and only skip it on Android 11.

Maybe that's the more responsible thing to do anyway....

File: app/src/main/java/mozilla/lockbox/LockboxAutofillService.kt [hunk 1]
```diff
@@ -234,4 +234,23 @@ class LockboxAutofillService(
             }
             .addTo(compositeDisposable)
     }
+
+    /**
+     * Handles a fill failure safely.
+     *
+     * Passing null into a `fail` callback on Android 11 displays a "null" toast to the
+     * user.  This is due to a bug in Android, and until it is fixed we noop on null
+     * instead.
```
Comment (eliserichards) [inline app/src/main/java/mozilla/lockbox/LockboxAutofillService.kt:243]: Can you add the link to the google issue in this comment?

**Review by eliserichards — CHANGES_REQUESTED** [review]
Looks awesome! Just one nit and this can be merged.

**Review by eliserichards — APPROVED** [review]
Great job! 🔥

**Comment by Webbeh** [comment]
Great ! Just noticed the issue came from Lockwise.

Any ETA on it being merged ?

**Comment by Webbeh** [comment]
https://github.com/mozilla-lockwise/lockwise-android/issues/1219#issuecomment-727525094
Self-signed APK with the fix implemented for anyone willing to try and report bugs for that version.

**Comment by donm527** [comment]
Thank you for the reply and offering a solution. So, if I am reading correctly, this bug has been reported as far back as September and no official update of the app from Mozilla?? 

Thanks though and I may try the unsigned copy. When they do update through Google Play, will your copy update or do I have to uninstall and re-download from the Play Store? That would mean I'd have to keep an eye for updates on the store since your copy will not notify me in Play Store?

**Comment by Webbeh** [comment]
There is an official update that has been sent and is signed in the releases.
https://github.com/mozilla-lockwise/lockwise-android/releases/tag/release-v4.0.3

It's simply not on the play store for the sole reason that Google updated their play store minimum required API and this is still not 100% up to date.

The app is working well and after the fix, I've been using it since it came out.

**Comment by donm527** [comment]
Oh OK, great! Thanks for the clarity. 

Will this version I sideload see and update any future updates that appear in Play Store or will I have to uninstall and download the new version when I see one has been updated on the store?

TIA!

**Comment by donm527** [comment]
Will this version I sideload see and update any future updates that appear in Play Store or will I have to uninstall and download the new version when I see one has been updated on the store?

**Pull request merged**